ui/src/app/pages/access/access-renewal-page.spec.tsx

```diff
@@ -0,0 +1,169 @@
+    const newModules = fp.map(({moduleName, expirationEpochMillis}) => ({
+      moduleName,
+      expirationEpochMillis: moduleName === updateModuleName ? time : expirationEpochMillis
+    }), oldProfile.renewableAccessModules.modules);
+    const newProfile = fp.set(['renewableAccessModules', 'modules'], newModules, oldProfile)
+    profileStore.set({profile: newProfile, load, reload, updateCache});
+  }
+
+  function setCompletionTimes(completionFn) {
+    const {profile} = profileStore.get();
+    const newProfile = fp.flow(
+      fp.set('complianceTrainingCompletionTime', completionFn()),
+      fp.set('dataUseAgreementCompletionTime', completionFn()),
+      fp.set('publicationsLastConfirmedTime', completionFn()),
+      fp.set('profileLastConfirmedTime', completionFn())
+    )(profile)
+    profileStore.set({profile: newProfile,  load, reload, updateCache});
+  }
+
+  const profile = ProfileStubVariables.PROFILE_STUB;
+
+  let mockUpdateProfile: SpyInstance;
+
+  const component = () => {
+    return mount(<AccessRenewalPage/>);
+  };
+
+  const load = jest.fn();
+  const reload = jest.fn();
+  const updateCache = jest.fn();
+
+  beforeEach(() => {
+    const profileApi = new ProfileApiStub();
+
+    registerApiClient(ProfileApi, profileApi);
+    mockUpdateProfile = jest.spyOn(profileApi, 'updateProfile');
+
+    // mocking because we don't have access to the angular service
+    reload.mockImplementation(async() => {
+      profileStore.set({profile: profileStore.get().profile, load, reload, updateCache});
+    });
+
+    profileStore.set({profile, load, reload, updateCache});
+
+    serverConfigStore.set({config: {
+      enableDataUseAgreement: true,
+      gsuiteDomain: 'fake-research-aou.org',
+      projectId: 'aaa',
+      publicApiKeyForErrorReports: 'aaa',
+      enableEraCommons: true,
+      enableComplianceTraining: true
+    }});
+
+    const institutionApi = new InstitutionApiStub();
+    registerApiClient(InstitutionApi, institutionApi);
+  });
+
+
+  it('should render when the user is expired', async () => {
+    expireAllModules()
+    const wrapper = component();
+
+    setCompletionTimes(() => Date.now() - 1000 * 60 * 60 * 24 * EXPIRY_DAYS);
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesContainingText(wrapper, 'access has expired').length).toBe(1);
+    expect(findNodesByExactText(wrapper, 'Review').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Confirm').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'View & Sign').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Complete Training').length).toBe(1);
+  });
+
+  it('should show the correct state when profile is complete', async () => {
+    expireAllModules()
+
+    const wrapper = component();
+
+    udpateOneModule('profileConfirmation', oneYearFromNow());
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesByExactText(wrapper, 'Confirmed').length).toBe(1)
```

Comment:
please add comments to indicate which of these represent expected-complete and which are expected-incomplete